libvast/builtins/operators/from_read_load_parse.cpp

```diff
@@ -90,6 +90,18 @@ class parse_operator final : public crtp_operator<parse_operator> {
     return std::move(*parser);
   }
 
+protected:
+  auto infer_type_impl(operator_type input) const
+    -> caf::expected<operator_type> override {
+    if (input.is<chunk_ptr>()) {
+      return tag_v<table_slice>;
+    }
+    // TODO: Fuse this check with crtp_operator::instantiate()
+    return caf::make_error(ec::type_clash,
+                           fmt::format("'{}' does not accept {} as input",
+                                       to_string(), operator_type_name(input)));
+  }
```

Comment:
The additional `infer_type_impl` overrides are necessary currently, although @jachris mentioned a better approach, so we be able to get rid of this shortly.

Reply:
I think this is good for now, we can make the mentioned change later.